src/util/myKivaUtils.js

```diff
@@ -157,8 +157,12 @@ export const getIsMyKivaEnabled = (apollo, $kvTrackEvent, userPreferences, loanT
 		);
 
 		// Ensure that the user continues to see MyKiva after passing the loan limit
-		// Only update the user preferences if running client-side
-		if (isMyKivaExperimentEnabled && !hasSeenMyKiva && typeof window !== 'undefined') {
+		// Only update the user preferences if running client-side and defined
+		if (isMyKivaExperimentEnabled
+			&& !hasSeenMyKiva
+			&& typeof window !== 'undefined'
+			&& typeof userPreferences !== 'undefined'
```

Comment:
In what case is this `undefined`?

Reply:
A guest user could be in test group, mainly for this: https://kiva.atlassian.net/browse/MP-1409